lib/components.ex

```diff
@@ -0,0 +1,57 @@
+defmodule LiveSvelte.Components do
+  @moduledoc """
+  Macros to improve the developer experience of crossing the Liveview/Svelte boundary.
+  """
+
+  @doc """
+  Generates functions local to your current module that can be used to render Svelte components.
+  """
+  defmacro __using__(_opts) do
+    get_svelte_components()
+    |> Enum.map(&name_to_function/1)
+  end
+
+  @doc """
+  TODO: This could perhaps be optimized to only read the files once per compilation.
+  """
+  def get_svelte_components do
+    "./assets/svelte/"
+    |> Path.join("**/*.svelte")
+    |> Path.wildcard()
+    |> Enum.filter(&(not String.contains?(&1, "_build/")))
+    |> Enum.map(fn path ->
+      path
+      |> Path.basename()
+      |> String.replace(".svelte", "")
+    end)
+  end
+
+  defp name_to_function(name) do
+    quote do
+      def unquote(:"#{name}")(assigns) do
+        props =
+          assigns
+          |> Map.filter(fn
+            {:svelte_opts, _v} -> false
+            {k, _v} -> k not in [:__changed__]
+            _ -> false
+          end)
+
+        var!(assigns) =
+          assign(assigns,
+            __component_name: unquote(name),
+            props: props || %{}
+          )
+
+        ~H"""
+        <LiveSvelte.svelte
+          name={Map.get(var!(assigns), :__component_name)}
+          class={Map.get(var!(assigns), :class)}
+          ssr={LiveSvelte.get_ssr(var!(assigns))}
+          props={Map.get(var!(assigns), :props, %{})}
```

Comment:
I think the issue is with the changed assigns being passed into the generated components. 
In your `LiveSvelte.get_props/1` function you check only for the `:__changed__` key, but here I was filtering that out to make it work. If I use your function it breaks on first render, so I need to populate that `:__changed__` field I think.

Reply:
If you see a better way of structuring this code feel free to refactor. Originally I only had the `<LiveSvelte.render>` working, then added sigil support, and now your macro. It might be an idea to rethink some of the code inside `lib/component.ex` and make a better structure. Just throwing it out in case you see a better path forward.

Reply:
I managed to resolve this, turns out the `assign/2` function updates the `__changed__` key in assigns, which signals to Liveview that all call sites of that assign needs to be re-rendered - which makes sense. 
In my original attempt, I was calling `assign` on the `:name` key which is was causing a complete re-mount of the Svelte component. 
Using `Map.put_new/3` for the name field was all that was needed to prevent re-mounting!